app/services/decidim/repair_comments_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Decidim
+  class RepairCommentsService
+    include Decidim::TranslatableAttributes
+
+    def self.run
+      new.execute
+    end
+
+    def execute
+      return [] if ok?
+
+      update_comments!
+    end
+
+    def ok?
+      invalid_comments.empty?
+    end
+
+    def invalid_comments
+      @invalid_comments ||= Decidim::Comments::Comment.all.map do |comment|
+        next if translated_attribute(comment.body).is_a?(String)
+
+        comment.body.delete("machine_translations")
+        [comment, comment.body.values.first]
+      end.compact
```

Comment:
The way you are doing this is suboptimal, let's peer.

Reply:
For references: 

- https://www.rubyinrails.com/2017/11/16/use-find-each-instead-of-all-each-in-rails/